source/features/conversation-timeline-filter.tsx

```diff
@@ -0,0 +1,225 @@
+				<button type="reset" className="btn" data-close-dialog="">Cancel</button>
+			</div>
+		</form>
+	);
+
+	// This works on github enterprise - form is already preloaded
+	select('form', detailsDialog)?.remove();
+	// This works on normal github. Normally form is loaded in place of `include-fragment` after we open details dialog.
+	select('include-fragment', detailsDialog)?.remove();
+
+	detailsDialog.append(form);
+}
+
+function processTimelineItem(item: HTMLElement): void {
+	const pr = select('.js-comment[id^=pullrequestreview]', item);
+	const normalComment = select('.js-comment-container', item);
+
+	if (pr) {
+		processPR(item);
+	} else if (normalComment) {
+		applyDisplay(item, FilterSettings.ShowOnlyComments, FilterSettings.ShowOnlyUnresolvedComments);
+	} else {
+		applyDisplay(item, FilterSettings.ShowAll);
+	}
+}
+
+function processPR(item: HTMLElement): void {
+	let hasVisibleElement = false;
+
+	for (const threadContainer of select.all('.js-resolvable-timeline-thread-container', item)) {
+		const commentContainer = select('.inline-comment-form-container', threadContainer);
+
+		if (threadContainer.getAttribute('data-resolved') === 'true') {
+			applyDisplay(threadContainer, FilterSettings.ShowOnlyComments);
+		} else if (commentContainer === null) {
+			// There is 1 special case here when github shows you a comment that was added to previous comment thread but it does not show whether it is resolved or not resolved comment.
+			// It's kinda tricky to know what to do with this so it is marked as normal comment for meantime.
+			// We are just checking here if user is able to comment inside that timeline thread, if not then it means we have this special situation that was just described.
+			applyDisplay(threadContainer, FilterSettings.ShowOnlyComments, FilterSettings.ShowOnlyUnresolvedComments);
+		} else {
+			applyDisplay(threadContainer, FilterSettings.ShowOnlyUnresolvedComments, FilterSettings.ShowOnlyUnresolvedReviews, FilterSettings.ShowOnlyComments);
+		}
+
+		// We need to hide whole thread group if we have hidden all comments inside.
+		hasVisibleElement = hasVisibleElement || threadContainer.style.display === '';
+	}
+
+	if (hasVisibleElement) {
+		item.style.display = '';
+	} else {
+		item.style.display = 'none';
+	}
+}
+
+function applyDisplay(element: HTMLElement, ...displaySettings: FilterSettings[]): void {
+	if (displaySettings.includes(currentSettings) || currentSettings === FilterSettings.ShowAll) {
+		element.style.display = '';
+	} else {
+		element.style.display = 'none';
+	}
+}
+
+async function init(): Promise<any> {
+	await addTimelineItemsFilter();
+
+	// There are some cases when github will remove this filter. In that case we need to add it again.
+	// Example: Editing comment will make timeline filter to disappear.
+	observe(`#${timelineFiltersSelectorId}`, {
+		async remove() {
+			await addTimelineItemsFilter();
+		}
+	});
```

Comment:
There must be a way to merge this. Probably you can just observe the closest sidebar element instead
```suggestion
	observe(`#whatever sidebar`, {
		add() {
			void addTimelineItemsFilter();
		}
	});
```

Reply:
Done